crates/circuit/src/duration.rs

```diff
@@ -0,0 +1,60 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+
+#[pyclass(eq, module = "qiskit._accelerate.circuit")]
+#[derive(PartialEq, Clone, Copy, Debug)]
+#[allow(non_camel_case_types)]
+/// A length of time used to express circuit timing.
+///
+/// It defines a group of classes which are all subclasses of itself (functionally, an
+/// enumeration carrying data).
+///
+/// In Python 3.10+, you can use it in a match statement::
+///
+///   match duration:
+///      case Duration.dt(dt):
+///          return dt
+///      case Duration.s(seconds):
+///          return seconds / 5e-7
+///      case _:
+///          raise ValueError("expected dt or seconds")
+///
+/// And in Python 3.9, you can use ``isinstance`` to determine which variant
+/// is populated::
+///
+///   if isinstance(duration, Duration.dt):
+///       return duration[0]
+///   elif isinstance(duration, Duration.s):
+///       return duration[0] / 5e-7
+///   else:
+///       raise ValueError("expected dt or seconds")
```

Comment:
Minor nitpicking: I think it's easier to read if the `#[]` proc macros are immediately above the struct definition, rather than above the documentation - they're typically super important to know, and related to the definition.

Reply:
I wonder if there's a lint rule or formatting transform we could enable to catch this. I agree, and I still made the mistake lol

Reply:
I can't see one: https://rust-lang.github.io/rust-clippy/master/index.html.

I think it might be slightly tricky, since there's funniness around how the documentation might itself be generated by a called proc macro, and you can intermix `/// literal doc` and `#[doc = <something that results to a string>]`, but it would be nice if there was one.